crates/pop-cli/src/commands/build/spec.rs

```diff
@@ -440,7 +440,7 @@ impl BuildSpecCommand {
 		};
 
 		// Prompt the user for deterministic build only if the profile is Production.
-		let deterministic = if skip_deterministic_build {
+		let deterministic = if skip_deterministic_build || !prompt {
```

Comment:
If the user indicates no additional changes to the chain spec, there's no need to build the deterministic runtime.